dependencies.yaml

```diff
@@ -98,26 +98,31 @@ dependencies:
               cuda: "11.2"
             packages:
               - cuda-version=11.2
-              - cudatoolkit
           - matrix:
               cuda: "11.4"
             packages:
               - cuda-version=11.4
-              - cudatoolkit
           - matrix:
               cuda: "11.5"
             packages:
               - cuda-version=11.5
-              - cudatoolkit
           - matrix:
               cuda: "11.8"
             packages:
               - cuda-version=11.8
+          - matrix:
+              cuda: "12.0"
+            packages:
+              - cuda-version=12.0
```

Comment:
@jakirkham I moved this pin upward from below. I can explain the goal here: I split the `cudatoolkit` dependency list into _two_ lists. The first list pins `cuda-version=major.minor` based on the `cuda` matrix entry. The second list adds the relevant CUDA package names for each _major_ version: `cudatoolkit` for 11.* and `cuda-cudart-dev` for 12.*. This uses the new globbing feature from `rapids-dependency-file-generator` 1.8.0, and allows us to de-duplicate the previous code, which listed `cudatoolkit` under every CUDA 11 minor version rather than only once for all CUDA 11.* versions. Does that make more sense?